libkbfs/tlf_handle_resolve.go

```diff
@@ -756,7 +764,31 @@ func parseTlfHandleLoose(
 			ctx, t, []resolvableUser{rit}, nil, nil, idGetter)
 		if err == nil && iteamHandle.tlfID != tlf.NullID {
 			// The iteam already has a TLF ID, let's use it.
-			return iteamHandle, nil
+
+			extensionList := tlf.HandleExtensionList(extensions)
+			sort.Sort(extensionList)
+			_, finalizedInfo := extensionList.Splat()
+
+			if finalizedInfo == nil && idGetter != nil {
+				// Implicit team chat migration might have set the ID to an
+				// old folder that has since been reset.  Check with the
+				// server to see if that has happened, and if so, don't use
+				// that ID.  When we migrate existing TLFs to iteams, we can
+				// probably override the sigchain link with a new one
+				// containing the correct TLF ID.
+				latestHandle, err := idGetter.GetLatestHandleForTLF(
```

Comment:
I forgot how often `parseTlfHandleLoose` is called; is this gonna slow down the happy path?

Reply:
This is on the path every time a TLF name is resolved into a handle, and it will slow things down with a round-trip to the mdserver in the case where there's an implicit team corresponding to the TLF, and that team has a TLF ID in its sigchain.  I'm not sure how often that will be the case -- until a few days ago I thought it wouldn't happen until we turned on iteam migration for KBFS, but I was wrong.

I will spend a few minutes thinking about whether I can cache this effectively somehow.